### PR TITLE
Make plug-in loading webpack-friendly

### LINKED ISSUE #3: Plug-in loading is not Webpack-friendly
I am basing a server on this framework and have a need to [bundle it with Webpack](https://webpack.js.org).
This runs into problems in the loading of core plug-in modules that are loaded by mapping a list of module names through the `require` function.
Consequently, the actual module dependencies are obscured and Webpack cannot see them to transform them.

Better would be to `require` literal module names.

### PR DESCRIPTION
**Pull request by cdamus**
- only require literal module names

Fixes #3.

Contributed on behalf of STMicroelectronics.

### CODE REVIEW

File: packages/modelserver-node/src/di.ts [hunk 1]
```diff
@@ -33,11 +33,10 @@ export async function createContainer(modelServerPort: number, loggingLevel: Log
 }
 
 async function loadModules(): Promise<ContainerModule[]> {
-    const modules = ['./logging-module', './server-module', './routes/routing-module'];
+    const modules = [require('./logging-module'), require('./server-module'), require('./routes/routing-module')];
```
Comment (eneufeld) [inline packages/modelserver-node/src/di.ts:36]: why do we need the require here?

**Review by eneufeld — COMMENTED** [review]
Can't we use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import#dynamic_imports instead of require?
@sdirix what is your opinion on this?

**Comment by cdamus** [comment]
> Can't we use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import#dynamic_imports instead of require?

It didn't seem useful because all of the plug-ins available must be loaded and given an opportunity to register providers before the server is ready to take requests. I'm not sure what the benefit of a dynamic imports is in this case.

**Comment by eneufeld** [comment]
> > Can't we use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import#dynamic_imports instead of require?
> 
> It didn't seem useful because all of the plug-ins available must be loaded and given an opportunity to register providers before the server is ready to take requests. I'm not sure what the benefit of a dynamic imports is in this case.

To my understanding `require('./logging-module')` is a dynamic loading or am I mistaken?

**Comment by cdamus** [comment]
> To my understanding `require('./logging-module')` is a dynamic loading or am I mistaken?

Ah, I see. We could just statically `import * as Xyz from '@whatever/xyz'` them, you mean? If that's preferred, I can do that.

**Comment by cdamus** [comment]
Yeah. I did it this way because it's exactly how the Theia code generator does it, so Theia peeps should see something familiar 😉

**Comment by sdirix** [comment]
The import statement will be translated into a `require`

> Can't we use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import#dynamic_imports instead of require? @sdirix what is your opinion on this?

Require is a sync import while dynamic imports are async, so they can't be interchanged by default. AFAIK dynamic imports within Node also require the imported module (or the importer?) to be an ESM module and Node to be of at least version 14 (or 16?) and correspondingly configured to being able to handle ESM modules. At least when used directly, when it's transspiled to some other mechanic/pattern (e.g. a "deferred require") via Webpack then of course this doesn't need to be true anymore.

I think Christian's code is fine for the Node world, it's not an uncommon pattern. Personally I would stick with normal `import` statements within Typescript files as long as I don't need any of the additional functionality provided by `require`. These `import` statements are then transspiled to `require` anyway.

**Pull request merged**